### PR TITLE
Eg 77 effectiveness label moves

### LINKED ISSUE #77: Effectiveness label for Pokemon moves
**Is your feature request related to a problem? Please describe.**
When the user goes to the versus screen, we want to show them the effectiveness of their Pokemon versus the enemy Pokemon.

**Describe the solution you'd like**
In the moves list, we'll be adding a "very effective" label to certain moves depending on enemy Pokemon type and weaknesses.

### PR DESCRIPTION
**Pull request by BlackCubes**
## Changes
1. Created a getPokeMoveType function that gets the type object from the pokemon moves API in api.js.
2. Created getPokeMoveTypeAPI function that fulfils the promise of getPokeMoveType in App.js.
3. Created enemyPokeMoveTypes states in App.js.
4. Passed enemyPokeMoveTypes state to Main component as prop in App.js.
5. Passed enemyPokeMoveTypes prop to EnemyPokemon component as prop in Main.js.
6. Created friendlyPokeType states in App.js.
7. Passed friendlyPokeType state to Main component as prop in App.js.
8. Passed friendlyPokeType prop to EnemyPokemon as prop in Main.js.
9. Created a moveEffectiveness function that checks to see if the moves of the enemy are effective against the friendly pokemon in EnemyPokemon.js.
10. Created a useEffect that fires the moveEffectiveness function depending if both enemyPokeMoveTypes and friendlyPokeType are both present in EnemyPokemon.js.
11. Passed effectivenessArray state to MovesList component as prop in EnemyPokemon.js.
12. Rendered the effectivenessArray in MovesList.js.

## Purpose
The enemy pokemon move's list should show whether it is "Super Effective" or "Ineffective" against the friendly pokemon.

## Approach
In order to show whether the enemy's pokemon move's is effective or not, the first step is to create an asynchronous function in `api.js` (`getPokeMoveType`) which does a `Promise.all` on the given moves array and store it in `enemyPokeMoveTypes` state. This was done in order for the new array from `getPokeMoveType` to be ordered sequentially with the enemy pokemon's moves array for display.

The next step was get the friendly pokemon's type, put it in a state (`friendlyPokeType` in `App.js`), and pass it all the way down to `EnemyPokemon.js`. This is so that friendly pokemon's type could be extracted from the custom `typeJson`.

Next, a function called `moveEffectiveness` was created that checks to see if the moves of the enemy are effective against the friendly pokemon in EnemyPokemon.js.

Overall, this creates a better UI experience for the user.

Closes #77

### CODE REVIEW

File: pokemon-app/src/components/EnemyPokemon.js [hunk 1]
```diff
@@ -1,18 +1,103 @@
...
 import Types from "./Types";
 import "../styles/EnemyPokemon.css";
 import "../styles/EnemyPokemon-mobile.css";
 
-const EnemyPokemon = ({ enemyPokemon, hasEnemy, enemyPokeSoundUrl }) => {
+/**
+ * Checks to see if the moves of the enemy are super effective, ineffective, or neutral to the friendly pokemon based on the friendly's type(s).
+ * @param {[object]} friendlyPokeType
+ * @param {[object]} enemyPokeMoveTypes
+ * @returns {[string]} An array of strings.
```
Comment (Lap343) [inline pokemon-app/src/components/EnemyPokemon.js:15]: love seeing more documentation

**Review by connielion — APPROVED** [review]
Very nice code! Keep up the great work! And man, Toad is OP.

**Review by Lap343 — APPROVED** [review]
Nice job with this! Now the styling

File: pokemon-app/src/components/MovesList.js [hunk 2]
```diff
@@ -47,7 +53,15 @@ const MovesList = ({ pokemonMoves, hasEnemy, statsOnTop, isEnemyPokemon }) => {
       >
         <ol id={`${isEnemyPokemon ? "enemy-" : ""}moves-list`}>
           {pokemonMoves.map((moveData, moveIndex) => (
-            <li key={moveIndex}>{moveData.move.name}</li>
+            <li key={moveIndex}>
+              {moveData.move.name}
+
+              {effectivenessArray && effectivenessArray.length && (
```
Comment (kmausolf) [inline pokemon-app/src/components/MovesList.js:59]: You should be able to shorten this to `effectivenessArray?.length && (`

File: pokemon-app/src/components/EnemyPokemon.js [hunk 3]
```diff
@@ -1,18 +1,103 @@
...
+  return moveEffectivenessCounter.map((effectiveCounter) => {
+    if (effectiveCounter >= 2) {
+      return "Super Effective";
+    } else if (effectiveCounter <= 0) {
+      return "Ineffective";
+    }
+
+    return "";
+  });
+};
```
Comment (krferrell) [inline pokemon-app/src/components/EnemyPokemon.js:77]: This is really cool. Just to check my understanding, this function will check whether a pokemon's move is effective against another pokemon based on their type? Nice work!

**Review by krferrell — APPROVED** [review]
Awesome and clean code Elias. Keep up the great work.

File: pokemon-app/src/components/MovesList.js (hunk 2 above)
Comment (BlackCubes) [reply pokemon-app/src/components/MovesList.js:59]: It's because the `MovesList` component is in both `SinglePokemon` and `EnemyPokemon` components. Since `SinglePokemon` doesn't render that array, I was worried that it would throw me an error since I wouldn't pass anything to that prop for `SinglePokemon`. Unless it wouldn't throw an error?

File: pokemon-app/src/components/EnemyPokemon.js (hunk 3 above)
Comment (BlackCubes) [reply pokemon-app/src/components/EnemyPokemon.js:77]: Yas!
